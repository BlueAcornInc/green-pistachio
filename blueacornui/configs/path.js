/**
 * Copyright Â© 2016 Magento. All rights reserved.
 * See COPYING.txt for license details.
 */

'use strict';

/**
 * Define paths.
 */

var webroot = '../webroot/';

module.exports = {
    binMage: webroot + 'bin/magento',
    pub: webroot + 'pub/static/',
    code: webroot + 'app/code/',
    design: webroot + 'app/design/',
    tmpLess: webroot + 'var/view_preprocessed/less/',
    tmpSource: webroot + 'var/view_preprocessed/source/',
    tmp: webroot + 'var',
    css: {
        setup: webroot + 'setup/pub/styles',
        updater: webroot + '../magento2-updater/pub/css'
    },
    less: {
        setup: webroot + 'app/design/adminhtml/Magento/backend/web/app/setup/styles/less',
        updater: webroot + 'app/design/adminhtml/Magento/backend/web/app/updater/styles/less'
    },
    uglify: {
        legacy: webroot + 'lib/web/legacy-build.min.js'
    },
    doc: webroot + 'lib/web/css/docs',
    spec: webroot + 'dev/tests/js/spec',
    static: {
        dir: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files',
        whitelist: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files/whitelist/',
        blacklist: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files/blacklist/',
        tmp: webroot + 'validation-files.txt'
    },
    webroot: '../webroot/'
};
