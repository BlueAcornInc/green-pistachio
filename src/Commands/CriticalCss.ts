import CriticalCssTask from '../Gulp/Tasks/CriticalCss';
import { CommandInterface, CommandOptionsInterface } from "./CommandInterface";

export interface CriticalCssCommandOptions extends CommandOptionsInterface {
    watch: boolean;
}

export default class CriticalCss implements CommandInterface {
    public async run(options: CriticalCssCommandOptions) {
        const { watch, project, theme } = options;
        const matchedTheme = project.getThemes().find(projectTheme => projectTheme.getData().path === theme);

        const criticalCss = new CriticalCssTask();
        if (watch) {
            criticalCss.watch(project, matchedTheme)(() => {});
        } else {
            criticalCss.execute(project, matchedTheme)(() => {});
        }

        return false;
    }
}
