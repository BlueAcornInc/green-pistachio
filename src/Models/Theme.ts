import glob from 'fast-glob';
import debug from 'debug';
const logger = debug('gpc:theme');

type ThemeConstructorArgs = ThemeData & {
    sourceDirectory: string;
    parent?: Theme;
};

// type CriticalCssInput = {
//     filepath: string;
//     urls: string[];
// };

export type ThemeData = {
    area: "frontend" | "adminhtml";
    path: string;
    enabled?: boolean;
    locales?: string[];
    stylesheets?: string[];
    // criticalCss?: CriticalCssInput[];
};

export default class Theme {
    private sourceDirectory: string;
    private parent?: Theme;
    private enabled: boolean | undefined;
    private data: ThemeData;

    constructor(config: ThemeConstructorArgs) {
        const {
            sourceDirectory,
            area,
            path,
            parent,
            locales,
            stylesheets,
            // criticalCss,
            enabled
        } = config;

        this.sourceDirectory = sourceDirectory;
        this.parent = parent;
        this.enabled = enabled || false;
        this.data = {
            area,
            path,
            locales,
            stylesheets,
            // criticalCss
        };
    }

    public configure(data: Partial<ThemeData>) {
        this.data = {
            ...this.data,
            ...data
        };
    }

    public getSourceDirectory() {
        return this.sourceDirectory;
    }

    public getParent() {
        return this.parent;
    }

    public getData() {
        return this.data;
    }

    public getLocales(): string[] {
        if (this.data.locales) {
            return this.data.locales;
        }

        logger(`${this.data.path}: Can't Determine Locales, defaulting to en_US`);

        return ['en_US'];
    }

    public getStyleSheets(): string[] {
        if (this.data.stylesheets) {
            return this.data.stylesheets;
        }

        logger(`${this.data.path}: Stylesheets not explicitly defined, detecting entries now.`);

        const files: Set<string> = new Set();

        const discoveredFiles = glob
            .sync(`${this.getSourceDirectory()}/**/[^_]*.less`)
            .map(sourceFile => sourceFile.replace(`${this.getSourceDirectory()}/web/`, ''));

        for (const file of discoveredFiles) {
            files.add(file);
        }

        let childTheme = this.getParent();

        while (childTheme) {
            for (const file of childTheme.getStyleSheets()) {
                files.add(file);
            }
            childTheme = childTheme.getParent();
        }

        const stylesheets = Array.from(files);

        logger(`${this.data.path}: Discovered the following stylesheets: ${stylesheets.join(', ')}`);

        return stylesheets;
    }

    // public getCriticalPaths(): CriticalCssInput[] {
    //     if (this.data.criticalCss) {
    //         return this.data.criticalCss;
    //     }

    //     logger(`No critical path data available for: ${this.data.path}`);

    //     return [];
    // }

    public getEnabled() {
        return this.enabled;
    }

    public setEnabled(enabled: undefined | boolean) {
        this.enabled = enabled;
    }
}
