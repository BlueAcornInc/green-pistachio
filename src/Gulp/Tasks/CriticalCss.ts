import { watch, TaskFunction, parallel } from "gulp";
import puppeteer from "puppeteer";
import { promises as fs } from 'fs';
import { join, dirname } from 'path';
import critical from 'critical';
import purge from 'css-purge';
import debug from 'debug';
import Project from "../../Models/Project";
import { TaskInterface } from "./TaskInterface";
import glob from "fast-glob";
const logger = debug('gpc:gulp:criticalCss');

export default class CriticalCss implements TaskInterface {
    private firstRun: boolean = true;
    private cacheMap: Map<string, boolean> = new Map();

    /**
     * Default options to be passed into the critical css node module
     */
    public static options = {
        dimensions: [
            {
                width: 375,
                height: 900
            },
            {
                width: 1440,
                height: 1080
            }
        ],
        minify: true
    };

    /**
     * Creates a critical.css file by project configuration
     * 
     * @param project | Project
     * @param theme | Theme
     * @returns TaskFunction
     */
    execute(project: Project): TaskFunction {
        const themes = project.getThemes();

        const tasks: TaskFunction[] = themes.map(theme => {
            const task: TaskFunction = async (done) => {
                await Promise.all(
                    theme.getCriticalPaths().map(async criticalPath => {
                        const cssFromUrls = await Promise.all(
                            criticalPath.urls.map(async url => {
                                try {
                                    const css = await this.getCriticalCssByUrl(project, url);
    
                                    return css;
                                } catch (err) {
                                    logger(`Error getting CSS from ${url} for ${theme.getData().path}: ${err}`);
    
                                    return '';
                                }
                            })
                        );
    
                        const css = cssFromUrls.reduce((allCss: string, cssForUrl) => allCss + cssForUrl, '');
    
                        const purgedCss = await new Promise<string>(resolve => {
                            purge.purgeCSS(css, {
                                trim: true,
                                shorten: true,
                                css: ''
                            }, (err, result) => {
                                if (err) {
                                    logger(`Error purging CSS from critical path: ${criticalPath.filepath}`);
                                    logger(`The critical css for ${criticalPath.filepath} may have duplicate styles depending on provided configuration.`);
                                    resolve(css);
                                } else {
                                    resolve(result);
                                }
                            });
                        });
    
                        try {
                            // Delete old file if it exists
                            await fs.unlink(criticalPath.filepath);
                        } catch (err) {}
    
                        await fs.mkdir(dirname(criticalPath.filepath), { recursive: true });
                        await fs.writeFile(criticalPath.filepath, purgedCss);
                    })
                );
                done();
            };

            task.displayName = `critcalCss<${theme.getData().path}>`;

            return task;
        });

        if (tasks.length === 0) {
            logger(`No Critical CSS Tasks Running`);
            tasks.push(done => done());
        }

        return parallel(...tasks);
    }

    /**
     * Watch task
     * 
     * @param project | Project
     * @param theme | Theme[]
     * @returns TaskFunction
     */
    watch(project: Project): TaskFunction {
        return () => {
            watch(
                this.getWatchFiles(project),
                this.execute(project)
            )
        };
    }

    /**
     * Get watch files that retrigger critical css compilation
     * 
     * @param project | Project
     * @param themes | Themes[]
     * @returns string[]
     */
    private getWatchFiles(project: Project) {
        let files: string[] = [];

        for (const theme of project.getThemes()) {
            for (const publicDirectory of project.getThemePubDirectories(theme)) {
                files = files.concat(
                    theme.getStyleSheets()
                        .map(
                            stylesheet => join(
                                publicDirectory,
                                stylesheet.replace('.less', '.css')
                            )
                        )
                )
            }
        }

        return files;
    }

    /**
     * Get critical css string from the given URL
     * 
     * @param project | Project
     * @param url | string
     * @returns string
     */
    private async getCriticalCssByUrl(project: Project, url: string) {
        const content = await this.getContentByUrl(url);
        const userConfig = {};
        project.hooks.gulp.criticalCssConfig.call(userConfig);

        return new Promise((resolve, reject) => {
            critical.generate({
                html: content,
                ...CriticalCss.options,
                ...(userConfig || {})
            }, (err, { css }) => {
                if (err) {
                    logger(err);
                    reject(err);
                    return;
                }

                resolve(css);
            });
        });
    }

    /**
     * Get page content by url, cache using filesystem as puppeteer process is quite slow
     * 
     * @param url | string
     * @returns string
     */
    public async getContentByUrl(url: string) {
        const encodedUrl = Buffer.from(url, 'binary').toString('base64');
        const cacheDirectory = join(__dirname, '.css-critical-cache');

        if (this.firstRun) {
            const entries = await glob(`${cacheDirectory}/**/*`);

            await Promise.all(
                entries.map(
                    async entry => {
                        try {
                            await fs.unlink(entry);
                        } catch (err) {
                            logger(`Unable to delete: ${entry}`);
                        }
                    }
                )
            )

            this.firstRun = false;
        }


        if (!this.cacheMap.has(url)) {
            const browser = await puppeteer.launch();
            const page = await browser.newPage();

            await page.goto(url, {
                waitUntil: 'networkidle2'
            });
            await page.waitForTimeout(10000);

            const content = await page.content();

            await fs.writeFile(
                join(
                    __dirname,
                    encodedUrl
                ),
                content
            );

            this.cacheMap.set(url, true);

            return content;
        } else {
            const cachedBuffer = await fs.readFile(
                join(
                    __dirname,
                    encodedUrl
                )
            );

            return cachedBuffer.toString();
        }
    }
}