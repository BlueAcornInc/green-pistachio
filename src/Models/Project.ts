import { join } from 'path';
import { SyncHook } from 'tapable';
import debug from 'debug';
import Module from "./Module";
import Theme, { ThemeData } from "./Theme";
import ConfigLoader from './Project/ConfigLoader';
import { Configuration } from 'webpack';
import WebpackConfigFactory from '../Webpack/ConfigFactory';
const logger = debug('gpc:project');

type ProjectConstructorArgs = {
    root: string;
    themes: Theme[];
    modules: Module[];
    enabledThemes?: string[];
};

type WebpackDevServerConfig = {
    proxyUrl: string;
    devServerUrl: string;
    devServerPort: number;
};

export default class Project {
    public root: string;
    private themes: Theme[];
    private enabledThemes: Theme[];
    private modules: Module[];
    private webpackDevelopmentMode: boolean = false;
    private proxyUrl: string = '';
    private devServerConfig: Partial<WebpackDevServerConfig> = {};

    public hooks = {
        // These hooks allow for modifying various gulp configurations
        gulp: {
            babelConfig: new SyncHook(["config"]),
            babelTypeScriptConfig: new SyncHook(["config"]),
            pngSpriteConfig: new SyncHook(["config"]),
            svgSpriteConfig: new SyncHook(["config"]),
            eslintConfig: new SyncHook(["config"]),
            criticalCssConfig: new SyncHook(["config"]),
        },
        webpack: {
            config: new SyncHook(["config"]),
            entryGlobs: new SyncHook(["globs"]),
            appendConfig: new SyncHook<[Configuration[], WebpackConfigFactory]>(["webpackConfigs", "configFactory"])
        },
        configure: new SyncHook(["project"])
    };

    public experiments = {
        webpack: {
            // Set to true in the the project.configure hook to enable unstable HMR functionality
            hmr: false,
            // Set to true to enable css modules
            cssModules: false,
            // Set to true to emit files to the pub directory instead of the individual theme directories
            emitFilesToPub: false,
        }
    };

    constructor(config: ProjectConstructorArgs) {
        const { root, themes, modules, enabledThemes } = config;

        this.root = root;
        this.themes = themes;
        this.modules = modules;
        this.enabledThemes = enabledThemes
            ? themes.filter(theme => enabledThemes.includes(theme.getData().path))
            : [];
    }

    public configure() {
        if (this.enabledThemes.length > 0) {
            // Enable default themes
            for (const theme of this.enabledThemes) {
                theme.setEnabled(true);
            }
        } else {
            for (const theme of this.themes) {
                if (
                    !theme
                        .getSourceDirectory()
                        .includes(
                            join(
                                this.getRootDirectory(),
                                'vendor'
                            )
                        )
                ) {
                    theme.setEnabled(true);
                    this.enabledThemes.push(theme);
                }
            }
        }

        // Enable default modules
        for (const magentoModule of this.modules) {
            if (
                !magentoModule
                    .getSourceDirectory()
                    .includes(
                        join(
                            this.getRootDirectory(),
                            'vendor'
                        )
                    )
            ) {
                magentoModule.setEnabled(true);
            }
        }

        this.hooks.configure.call(this);
    }

    public configureTheme(themeData: Partial<ThemeData>) {
        if (!themeData.path) {
            logger(`can not configure theme without defined path: ${JSON.stringify(themeData)}`);
            return;
        }

        const theme = this.themes.find(theme => theme.getData().path === themeData.path);

        if (!theme) {
            logger(`can not find theme with path = '${themeData.path}'`);
            return;
        }

        theme.configure(themeData);

        if (themeData.hasOwnProperty('enabled')) {
            theme.setEnabled(themeData.enabled);
        }
    }

    public enableModule(moduleName: string) {
        const magentoModule = this.modules.find(magentoModule => magentoModule.getName() === moduleName);

        if (!magentoModule) {
            logger(`could not find module by name: ${moduleName}`);
            return;
        }

        magentoModule.setEnabled(true);
    }

    public getThemes() {
        return this.enabledThemes.filter(magentoTheme => magentoTheme.getEnabled());
    }

    public getAllThemes() {
        return this.themes;
    }

    public getModules() {
        return this.modules.filter(magentoModule => magentoModule.getEnabled());
    }

    public getAllModules() {
        return this.modules;
    }

    public getRootDirectory() {
        return this.root;
    }

    public getThemePubDirectories(theme: Theme) {
        return theme.getLocales().map(locale => join(
            this.getStaticDirectory(),
            theme.getData().area,
            theme.getData().path,
            locale
        ));
    }

    public getWebpackOutputDirectories(theme: Theme) {
        if (this.isWebpackDevelopmentMode() || this.experiments.webpack.emitFilesToPub) {
            return theme.getLocales().map(locale => join(
                this.getRootDirectory(),
                ...(
                    this.isWebpackDevelopmentMode()
                    ? ['static']
                    : ['pub', 'static']
                ),
                theme.getData().area,
                theme.getData().path,
                locale
            ));
        }

        return [join(theme.getSourceDirectory(), 'web')];
    }

    public getStaticDirectory() {
        return join(
            this.getRootDirectory(),
            'pub',
            'static'
        );
    }

    public isWebpackDevelopmentMode() {
        return this.webpackDevelopmentMode;
    }

    public setWebpackDevelopmentMode() {
        this.webpackDevelopmentMode = true;
    }

    public setProxyUrl(url: string) {
        this.proxyUrl = url;
    }

    public getProxyUrl(): string {
        if (this.proxyUrl.length === 0) {
            if (!debug.enabled('gpc:logger')) {
                debug.enable('gpc:logger');
            }
            logger(`A project URL must be provided in your projects ${ConfigLoader.CONFIG_FILE}`);
            logger(`\n\nExample:
module.exports = project => {
project.hooks.configure.tap("Set Product Url", () => {
    project.setProxyUrl("https://example.test");
});
}
            `);

            throw new Error(`A project URL must be provided in your projects ${ConfigLoader.CONFIG_FILE}`);
        }

        return this.proxyUrl;
    }

    public setWebpackDevServerConfig(config: Partial<WebpackDevServerConfig>) {
        if (config.proxyUrl) {
            this.proxyUrl = config.proxyUrl;
        }

        this.devServerConfig = config;
    }

    public getWebpackDevServerConfig(): WebpackDevServerConfig {
        return {
            proxyUrl: this.getProxyUrl(),
            devServerUrl: 'green-pistachio.test',
            devServerPort: 8080,
            ...(this.devServerConfig || {})
        };
    }
}
