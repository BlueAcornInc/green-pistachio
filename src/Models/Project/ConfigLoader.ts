import { promises as fs } from 'fs';
import { join } from 'path';
import debug from 'debug';
import Project from "../Project";
const logger = debug('gpc:configLoader');

export default class ConfigLoader {
    public static CONFIG_FILE = 'green-pistachio.config.js';

    constructor(private project: Project) {}

    get filePath(): string {
        return join(
            this.project.getRootDirectory(),
            ConfigLoader.CONFIG_FILE
        );
    }

    public async configExists(): Promise<boolean> {
        try {
            const configExists = await fs.stat(this.filePath);

            if (configExists) {
                return true;
            }
        } catch (err) {
            logger(`Config file can't be found at ${this.filePath}`);
        }

        return false;
    }

    public async loadConfig(): Promise<boolean> {
        const configExists = await this.configExists();

        if (configExists) {
            try {
                this.interopRequire()(this.project);
                return true;
            } catch (err) {
                logger(`Problem occurred while running user provided configuration file: ${err}`);
            }
        }

        return false;
    }

    private interopRequire() {
        const configModule = require(this.filePath);
        return configModule && configModule.__esModule ? configModule.default : configModule;
    }
}