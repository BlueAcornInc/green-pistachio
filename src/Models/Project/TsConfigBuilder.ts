import { promises as fs } from 'fs';
import { join } from 'path';
import debug from 'debug';
import Project from "../Project";
const logger = debug('gpc:typescript:configBuilder');

export default class TsConfigBuilder {
    public getFilePath(): string {
        return join(__dirname, '..', '..', '..', 'green-pistachio.tsconfig.json');
    }

    public async configFileExists(): Promise<boolean> {
        try {
            await fs.stat(this.getFilePath());

            return true;
        } catch (err) {
            return false;
        }
    }

    public async emitConfigFile(project: Project) {
        logger(`Generating Projects tsconfig.json base`);
        const include: string[] = [];
        const paths: Record<string, string[]> = {}

        for (const module of project.getModules()) {
            const moduleSrcDir = module.getSourceDirectory().replace(project.getRootDirectory(), '');

            const modulePaths: string[] = [
                join(moduleSrcDir, 'view/frontend/web/ts/*'),
                join(moduleSrcDir, 'view/base/web/ts/*'),
                join(moduleSrcDir, 'view/adminhtml/web/ts/*'),
            ];

            paths[`${module.getName()}/*`] = modulePaths.map(modulePath => modulePath.replace(/^\//, ''));

            include.push(join(moduleSrcDir, 'view/frontend/web/ts/**/*'));
            include.push(join(moduleSrcDir, 'view/base/web/ts/**/*'));
            include.push(join(moduleSrcDir, 'view/adminhtml/web/ts/**/*'));
        }

        for (const theme of project.getAllThemes()) {
            const themeSrcDir = theme.getSourceDirectory().replace(project.getRootDirectory(), '');

            include.push(join(themeSrcDir, '**/*'));
            
            // Add theme overrides to module paths
            for (const moduleName of Object.keys(paths)) {
                paths[moduleName].unshift(
                    join(themeSrcDir, moduleName.replace(/\/\*$/, ''), 'web/ts/*').replace(/^\//, '')
                );
            }
        }

        logger(`Resolved include paths to: ${JSON.stringify(include)}`);
        logger(`Resolved module paths to: ${JSON.stringify(paths)}`);

        const writePath = this.getFilePath();
        const config = this.getBaseConfig();

        config.include = include.map(includePath => `./${includePath}`);
        config.compilerOptions.paths = paths;

        try {
            logger(`Writing config file: ${writePath}`);
            await fs.writeFile(writePath, JSON.stringify(config, null, 4));
        } catch (err) {
            logger(`Problem writing config file: ${writePath}`);
            throw err;
        }

        return writePath;
    }

    private getBaseConfig() {
        return {
            "include": [
              "./app/**/*"
            ],
            "compilerOptions": {
              "baseUrl": ".",
              "paths": {},
              "strict": true,
              "esModuleInterop": true,
              "lib": [
                "dom",
                "es2015"
              ],
              "jsx": "react",
              "emitDecoratorMetadata": true,
              "experimentalDecorators": true,
              "target": "es5",
              "allowJs": true,
              "skipLibCheck": true,
              "allowSyntheticDefaultImports": true,
              "forceConsistentCasingInFileNames": true,
              "module": "esnext",
              "moduleResolution": "node",
              "resolveJsonModule": true,
              "isolatedModules": true,
              "noEmit": true
            }
          }
    }
}