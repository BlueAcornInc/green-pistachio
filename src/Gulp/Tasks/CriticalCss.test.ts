/**
 * @jest-environment node
 */
import { join } from 'path';
import Project from '../../Models/Project';
import Theme from '../../Models/Theme';
import { promises as fs } from 'fs';
import CriticalCss from './CriticalCss';
 
describe('Gulp: Critical CSS', () => {
    it('should include only critical styles', async (done) => {
        const root = join(
            __dirname,
            '__fixtures__',
            'critical'
        );

        const project = new Project({
            themes: [
                new Theme({
                    sourceDirectory: `${root}/vendor/magento/theme-frontend-blank`,
                    area: 'frontend',
                    path: 'Magento/blank',
                    enabled: true,
                    criticalCss: [{
                        filepath: `${root}/vendor/magento/theme-frontend-blank/web/css/critical.css`,
                        urls: ['']
                    }]
                })
            ],
            modules: [],
            root,
        });

        const criticalCss = new CriticalCss();
        CriticalCss.options = {
            ...CriticalCss.options,
            // @ts-ignore
            base: root
        };
        // Trust that puppeteer runs properly, too slow for testing
        criticalCss.getContentByUrl = async () => `
            <link rel="stylesheet" type="text/css" media="all" href="styles.css"></link>
            <h1>Hello World</h1>
        `;
        criticalCss.execute(project)(async () => {
            const styles = await fs.readFile(`${root}/vendor/magento/theme-frontend-blank/web/css/critical.css`);
            expect(styles.toString()).toMatchSnapshot();
            try {
                await fs.unlink(`${root}/vendor/magento/theme-frontend-blank/web/css/critical.css`);
            } catch (err) {}
            done();
        });
    });
});