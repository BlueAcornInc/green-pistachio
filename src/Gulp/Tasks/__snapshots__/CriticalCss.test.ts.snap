// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gulp: Critical CSS should include only critical styles 1`] = `"h1{color:red}"`;
